### PR TITLE
feat: add initial setup & BKTConfig 

### PR DESCRIPTION
**Pull request by yshrsmz**
- use [pnpm](https://pnpm.io/) for package management
  - npm is ok, but I want more efficiency
  - yarn is another solution but pnpm is chosen by lots of popular packages these days
    - such as [Next.js](https://github.com/vercel/next.js/) and [Nuxt.js](https://github.com/nuxt/framework)
- use [vite](https://vitejs.dev) for building/bundling
  - I'm still evaluating it, so I may switch to a different tool if not suitable for this repo.

### CODE REVIEW

File: src/BKTConfig.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  fetch: FetchLike
+}
+
+export interface BKTConfig extends RawBKTConfig {
+  eventsFlushInterval: number
+  eventsMaxBatchQueueCount: number
+  pollingInterval: number
+}
+
+export const defineBKTConfig = (config: RawBKTConfig): BKTConfig => {
```
Comment (yshrsmz) [inline src/BKTConfig.ts:36]: instead of creating a builder class I've created this factory function for BKTConfig

Comment (cre8ivejp) [reply src/BKTConfig.ts:36]: Thanks for the PR!
Can you provide a sample code of how it will look when initializing the SDK?

Comment (yshrsmz) [reply src/BKTConfig.ts:36]: sure, here's a gist

https://gist.github.com/yshrsmz/dc3dbd0eeee878c584ef45532a74702e

Basically, it's going to look like the below:

```typescript
const config = defineBKTConfig({ ... })
const user = defineBKTUser({ ... })
const timeoutMillis = 5_000

initialize(config, user, timeoutMillis)
    // returns Promise<void>
    .then(() => {
      // initial fetch has completed
      const variation = getBKTClient().stringVariation(featureId, 'default-value')
      // do something with a variation value
    })
    .catch((e) => {
      // error while initializing/initial fetch
    })
```

File: package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@bucketeer/js-client-sdk",
```
Comment (yshrsmz) [inline package.json:2]: changed to `@bucketeer/js-client-sdk` as discussed.

**Review by cre8ivejp — APPROVED** [review]
Thank you!

**Comment by yshrsmz** [comment]
Thanks!

**Pull request merged**